File: bench/fixtures/needs/python/lazy.py
"""TRUE BUT HIDDEN: the dependency is resolved at runtime by name, so the
specifier is a string and the reader has no import statement to place.

Shape from .corpus/django-django/django/utils/module_loading.py:15 —
`module = import_module(module_path)`.
"""

from importlib import import_module


def handle(path: str) -> int:
    module = import_module("bench.fixtures.needs.python.model")
    return len(module.Request(path).path)

File: bench/fixtures/needs/typescript/lazy.ts
// TRUE BUT HIDDEN: the only reference to ./model is a dynamic import, whose
// specifier may be built at runtime. The dependency is real; the reader is
// documented to withhold rather than credit it.
//
// Shape from .corpus/vitejs-vite/packages/vite/src/node/cli.ts:227 -- a
// destructured await of a dynamic import, there naming a sibling module.

export async function handle(path: string): Promise<number> {
  const { makeRequest } = await import("./model");
  return makeRequest(path).path.length;
}
